Source/iOS/OIDExternalUserAgentCatalyst.m

```diff
@@ -0,0 +1,148 @@
+#import "OIDExternalUserAgentRequest.h"
+
+#if TARGET_OS_MACCATALYST
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OIDExternalUserAgentCatalyst ()<ASWebAuthenticationPresentationContextProviding>
+@end
+
+@implementation OIDExternalUserAgentCatalyst {
+  UIViewController *_presentingViewController;
+
+  BOOL _externalUserAgentFlowInProgress;
+  __weak id<OIDExternalUserAgentSession> _session;
+  ASWebAuthenticationSession *_webAuthenticationVC;
+}
+
+- (nullable instancetype)init {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
+  return [self initWithPresentingViewController:nil];
+#pragma clang diagnostic pop
+}
+
+- (nullable instancetype)initWithPresentingViewController:
+    (UIViewController *)presentingViewController {
+  self = [super init];
+  if (self) {
+    NSAssert(presentingViewController != nil,
+             @"presentingViewController cannot be nil on iOS 13");
+    _presentingViewController = presentingViewController;
+  }
+  return self;
+}
+
+- (BOOL)presentExternalUserAgentRequest:(id<OIDExternalUserAgentRequest>)request
+                                session:(id<OIDExternalUserAgentSession>)session {
+  if (_externalUserAgentFlowInProgress) {
+    // TODO: Handle errors as authorization is already in progress.
+    return NO;
+  }
+
+  _externalUserAgentFlowInProgress = YES;
+  _session = session;
+  BOOL openedUserAgent = NO;
+  NSURL *requestURL = [request externalUserAgentRequestURL];
+
+  __weak OIDExternalUserAgentCatalyst *weakSelf = self;
+  NSString *redirectScheme = request.redirectScheme;
+  ASWebAuthenticationSession *authenticationVC =
+      [[ASWebAuthenticationSession alloc] initWithURL:requestURL
+                                    callbackURLScheme:redirectScheme
+                                    completionHandler:^(NSURL * _Nullable callbackURL,
+                                                        NSError * _Nullable error) {
+    __strong OIDExternalUserAgentCatalyst *strongSelf = weakSelf;
+    if (!strongSelf) {
+        return;
+    }
+    strongSelf->_webAuthenticationVC = nil;
+    if (callbackURL) {
+      [strongSelf->_session resumeExternalUserAgentFlowWithURL:callbackURL];
+    } else {
+      NSError *safariError =
+          [OIDErrorUtilities errorWithCode:OIDErrorCodeUserCanceledAuthorizationFlow
+                           underlyingError:error
+                               description:nil];
+      [strongSelf->_session failExternalUserAgentFlowWithError:safariError];
+    }
+  }];
+      
+  authenticationVC.presentationContextProvider = self;
+  _webAuthenticationVC = authenticationVC;
+  openedUserAgent = [authenticationVC start];
+
+  if (!openedUserAgent) {
+    [self cleanUp];
+    NSError *safariError = [OIDErrorUtilities errorWithCode:OIDErrorCodeSafariOpenError
+                                            underlyingError:nil
+                                                description:@"Unable to open Safari."];
```

Comment:
May as well be explicit in the error message since there's only one, i.e. "ASWebAuthenticationSession"

Reply:
Done.